packages/dds/merge-tree/src/revertibles.ts

```diff
@@ -367,22 +419,27 @@ export function revertMergeTreeDeltaRevertibles(
 	driver: MergeTreeRevertibleDriver,
 	revertibles: MergeTreeDeltaRevertible[],
 ) {
+	let mergeTreeWithRevert: MergeTreeWithRevert | undefined;
+
 	while (revertibles.length > 0) {
 		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
 		const r = revertibles.pop()!;
 		const operation = r.operation;
-		switch (operation) {
-			case MergeTreeDeltaType.INSERT:
-				revertLocalInsert(driver, r);
-				break;
-			case MergeTreeDeltaType.REMOVE:
-				revertLocalRemove(driver, r);
-				break;
-			case MergeTreeDeltaType.ANNOTATE:
-				revertLocalAnnotate(driver, r);
-				break;
-			default:
-				unreachableCase(operation);
+		if (r.trackingGroup.size > 0) {
```

Comment:
Is this check (and the check in appendToMergeTreeDeltaRevertibles on deltaArgs.deltaSegments) specific to this refactor, or are they just useful checks to have?

Reply:
both. now we inspect the tracking groups to find the first tracked thing, which we use to find the root and the merge tree. if there is nothing to revert, we can't get the merge tree, but we don't need it either.